academic_observatory_workflows/workflows/oa_web_workflow.py

```diff
@@ -403,6 +407,81 @@ def to_dict(self) -> Dict:
         return dict_
 
 
+def get_institution_logo(ror_id: str, url: str, size: str, width: int, fmt: str, build_path) -> Tuple[str, str]:
+    """Get the path to the logo for an institution.
+    If the logo does not exist in the build path yet, download from the Clearbit Logo API tool.
+    If the logo does not exist and failed to download, the path will default to "/unknown.svg".
+
+    :param ror_id: the institution's ROR id
+    :param url: the URL of the company domain + suffix e.g. spotify.com
+    :param size: the image size of the small logo for tables etc.
+    :param width: the width of the image.
+    :param fmt: the image format.
+    :param build_path: the build path for files of this workflow
+    :return: The ROR id and relative path (from build path) to the logo
+    """
+    logo_path = f"/unknown.svg"
+
+    file_path = os.path.join(build_path, "logos", "institution", size, f"{ror_id}.{fmt}")
+    if not os.path.isfile(file_path):
+        clearbit_download_logo(company_url=url, file_path=file_path, size=width, fmt=fmt)
+    if os.path.isfile(file_path):
+        logo_path = make_logo_url(category="institution", entity_id=ror_id, size=size, fmt=fmt)
+
+    return ror_id, logo_path
+
+
+def get_wiki_description(titles: dict) -> List[tuple]:
+    """Get the wikipedia descriptions for the given titles.
+
+    :param titles: Dict with titles as keys and id's (from index table, ror_id or alpha3 country code) as values
+    :return: List with (id, wiki description) tuples
+    """
+    descriptions = []
+    titles_arg = []
+    for title, entity_id in titles.items():
+        entity_id = titles[title]
+        # Set description to NA for empty title
+        if title == "":
+            descriptions.append((entity_id, "NA"))
+        elif title == urllib.parse.unquote(title):
+            titles_arg.append(urllib.parse.quote(title))
+        else:
+            titles_arg.append(title)
+
+    response = requests.get(
+        f"https://en.wikipedia.org/w/api.php?action=query&format=json&prop=extracts"
+        f"&titles={'%7C'.join(titles_arg)}&redirects=1&exintro=1&explaintext=1"
+    )
+    if response.status_code != 200:
+        raise AirflowException()
+    response_json = response.json()
+    pages = response_json["query"]["pages"]
+
+    # Create mapping between redirected page title and original page title
+    redirects = {}
+    for title in response_json["query"].get("redirects", []):
+        redirects[title["to"]] = title["from"]
+
+    alpha_titles = {re.sub("[^A-Za-z]+", "", urllib.parse.unquote(k)): v for k, v in titles.items()}
+    for page_id, page in pages.items():
+        page_title = page["title"]
+        # Get page_title from redirect if it is present
+        page_title = redirects.get(page_title, page_title)
+
+        # Match entity_id and page_title
+        entity_id = alpha_titles[re.sub("[^A-Za-z]+", "", page_title)]
+
+        # Get description
+        try:
+            description = page["extract"]
```

Comment:
It would be good to add a function that cleans the Wikipedia descriptions, by removing text within brackets () and taking the first n sentences. I think you could use a regex to remove text within the brackets.

For example:
"Mali (/ˈmɑːli/ (audio speaker iconlisten); French pronunciation: ​[mali]), officially the Republic of Mali (French: République du Mali; Bambara: ߡߊߟߌ ߞߊ ߝߊߛߏߖߊߡߊߣߊ, romanized: Mali ka Fasojamana, Fula: 𞤈𞤫𞤲𞥆𞤣𞤢𞥄𞤲𞤣𞤭 𞤃𞤢𞥄𞤤𞤭, romanized: Renndaandi Maali, Arabic: جمهورية مالي), is a landlocked country in West Africa. Mali is the eighth-largest country in Africa, with an area of over 1,240,000 square kilometres (480,000 sq mi)."

Would become (when taking the first two sentences):
"Mali, officially the Republic of Mali, is a landlocked country in West Africa. Mali is the eighth-largest country in Africa, with an area of over 1,240,000 square kilometres."